### DISCUSSION TITLE
Limit the Scope

### DISCUSSION THREAD

**Discussion by odungern** [GfSE/CoCoML-Pages#7] [Ideas]
UML has been devised to specify software products. It is also used to generate software and thus offers many language features towards that end. SysML is a UML Profile and as such shares many characteristics. However, much less detail is needed to communicate the overarching concept of a system. 

So, the first step is to select the UML/SysML elements to use for that purpose. 

Three categories are proposed:
1. Well defined element to transform, e.g. _uml:Generalization_ or _uml:Composition_.
2. Weakly defined element with a name to transform, e.g. _uml:Association_. The name from a controlled vocabulary adds meaning.
3. Element not to transform. If used, a warning shall be issued.

**Comment by odungern** [comment]
Shall CoCoML be closed (with periodic updates) or must it allow user-defined extensions?

**Comment by odungern** [comment]
Better than specifying meaning of a relation with a _name_ is to assign a _stereotype_ such as «sysml:Allocate». But how to define such stereotypes for CoCoML?

**Comment by haarer** [comment]
I think it ist necessary to separate concepts from a representation.

Association has no more conceptual meaning than "this has Something to do with that", with added multiplicities.

Some semantics are derived from the Kind of associated Element, e.g. actor and use Case.

I agree, that stereotypes are necessary to Nail down the semantics, but this does require Prior Work in concepts.

That is the reason why [SAF](HTTPS://GitHub.com/gfse/saf-specification) has a conceptual model and any Profile Elements of SAF must trace to this conceptual model.
Even If SAF concepts are implemented by plain UML or sysml Elements, they are traced in order to have no specification holes.
